frontend/src/metabase/containers/ItemPicker/ItemPicker.tsx

```diff
@@ -100,6 +102,11 @@ function ItemPicker<TId>({
   const isPickingNotCollection = models.some(model => model !== "collection");
 
   const openCollection = collectionsById[openCollectionId];
+  const isOpenCollectionInPersonalCollection = openCollection?.is_personal;
+  const hideItems =
+    !searchString &&
+    showOnlyPersonalCollections &&
+    !isOpenCollectionInPersonalCollection;
```

Comment:
We want to still enable fetch when `searchString` is not null, meaning users are searching for dashboards.
![image](https://github.com/metabase/metabase/assets/1937582/12da0480-f8ba-457b-bf42-153a63cf4407)

But we don't want to fetch anything when opening a public collection (only root collection in this case), and is selecting a dashboard for a question in a personal collection.
![image](https://github.com/metabase/metabase/assets/1937582/e8633872-cd22-4044-813c-1139e05fbef7)